src/getFilePathforAbsRoot.ts

```diff
@@ -0,0 +1,23 @@
+import { getFilePath } from 'hono/utils/filepath'
+
+/**
+ * wrapper for getFilePath, with absolute root path
+ */
+export function getFilePathforAbsRoot(options: {
```

Comment:
I think that `getFilePathForAbsRoot` would be a better name.

Reply:
I agree.

I am considering a better name but have not yet been able to come up with one.

---

The current getFilePath function has the following features:

1. the relative path to the parent should be undefind (`../a` => undefined)
2. it normalizes the path for KV (considering the relative path to KV root `./hoge/fuga` => `hoge/fuga`, `/hoo` => `hoo`)
3. it appends defaultDocument (`/hoo` => `/hoo/index.html`)

ref: https://github.com/honojs/hono/blob/fc73d5d105b65ae823184d33e67c8228aa1c2e02/src/utils/filepath.test.ts

I think that getFilePath has two roles: 
features 1 and 2 are for KV, and the rest are for general use to serve something. Is that correct?


Relative to the parent or absolute path doesn't work because of the features 1 and 2.
I've tested that serveStatic in Bun and Deno also doesn't work.

I believe it would be better to prepare another function specifically for KV that utilizes getFilePath and implements features for KV as follows:

```ts
function getFilePathForKV(options :FilePathOptions) { // or more better function name
  if (/(?:^|\/)\.\.(?:$|\/)/.test(options.filename)) return
  const path = getFilePath(options)
  // ./assets/foo.html => assets/foo.html
  return path.replace(/^\.?\//, '')
}
```


What are your thoughts on this? If you agree with this opinion, I will create a pull request for it.

Reply:
These are for both KV and other file system-based runtimes.

'1' is necessary to avoid issues such as directory traversal. In older versions of Bun, the URL path was not normalized, so there was a possibility that the URL path included `..`, which could lead to unexpected behavior. I think we can handle it in a different place, such as `serve-static` for Bun, but currently, we are handling it in `getFilePath()`.

'2' is to normalize the path to find the file. The file path emitted from `getFilePath()` will be used as shown in the link:

https://github.com/honojs/hono/blob/fc73d5d105b65ae823184d33e67c8228aa1c2e02/src/adapter/bun/serve-static.ts#L38

In KV as well, `/foo` will fall back to `/foo/index.html`. So, this is necessary for KV.